docker-compose.yml

```diff
@@ -12,6 +12,7 @@ services:
     environment:
       - PORT=4000
       - NODE_ENV=development
+    env_file: .env
```

Comment:
Would make sense to have the .env file sit outside of the repo directory?
Just thinking about how it would be lost if you ever blew away your local repo.

Reply:
It looks like nestjs is just using [dotenv](https://www.npmjs.com/package/dotenv) rather than a json configuration, but it loads the .env directly rather than having something like docker-compose populate the environment.

Reply:
I tried `config` and thought about using JSON Schema library, but ultimately, I just decided to keep it simple for now and use environment variables. I'm validating them using TypeScript and building a nested `config` object here: https://github.com/ssciolla/canvas-course-manager-next/blob/issue-23-lti/ccm_web/server/config.ts There's probably a way a better way to do this, but there are some advantages to the simplicity of environment variables and not having to worry about a `dotenv`, `config`, package or whatever. ROHQ uses a similar pattern, so I was partly inspired by that.

Reply:
> Would make sense to have the .env file sit outside of the repo directory?
> Just thinking about how it would be lost if you ever blew away your local repo.

I would like to discuss the ideal configuration approach at a later date, if that's okay. I rarely need to wipe out my repository, but there are probably scenarios I'm not thinking of where that would be nice. An approach that uses something like `dotenv` explicitly would probably be nice to allow for easy use outside of Docker (i.e. straight `node`/`npm`, local database instance). I'm going to mark this is as resolved for now, but if there's a serious concern still here, please re-open and state what you'd like to see.